tools/devops/automation/templates/mac/build.yml

```diff
@@ -0,0 +1,188 @@
+parameters:
+
+- name: statusContext
+  type: string 
+  default: '10.16' # default context, since we started dealing with iOS devices. 
+
+- name: clearProvisionatorCache
+  type: boolean
+  default: false
+
+# can depend on the pool, which is annoying, but we should keep it in mind
+- name: keyringPass
+  type: string
+
+steps:
+
+- checkout: self
+- checkout: maccore
+  persistCredentials: true  # hugely important, else there are some scripts that check a single file from maccore that will fail
+
+- bash: |
+    security set-key-partition-list -S apple-tool:,apple: -s -k $KEYCHAIN_PASS login.keychain
+  env:
+    KEYCHAIN_PASS: ${{ parameters.keyringPass }}
+  displayName: 'Remove security UI-prompt (http://stackoverflow.com/a/40039594/183422)'
+  condition: succeededOrFailed() # we do not care about the previous process cleanup
+  continueOnError: true
+
+- bash: cd $(System.DefaultWorkingDirectory)/xamarin-macios/ && git clean -xdf
+  displayName: 'Clean workspace'
+
+
+# download the packages that have been created, install them, later download the zip files that contain the already built
+# tests and execute them.
+
+- pwsh : |
+    gci env: | format-table -autosize -wrap
+  displayName: 'Dump Environment'
+
+# Uses the powershell cmdlet to remove known directories that are not needed and use space.
+# we might hit issue https://github.com/PowerShell/PowerShell/issues/9246, so we always do exit 0
+- pwsh: |
+    Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin-macios/tools/devops/automation/scripts/System.psm1
+    Clear-HD 
+    exit 0  
+  displayName: 'Disk cleanup'
+  timeoutInMinutes: 5
+  continueOnError: true
+
+# Use a cmdlet to check if the space available in the devices root system is larger than 50 gb. If there is not
+# enough space available it:
+# 1. Set the status of the build to error. It is not a failure since no tests have been ran.
+# 2. Set a comment stating the same as what was sent to the status.
+# 3. Cancel the pipeline and do not execute any of the following steps.
+- pwsh: |
+    cd $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin-macios/tools/devops/automation/scripts/
+    Import-Module ./System.psm1
+    Import-Module ./VSTS.psm1
+    Import-Module ./GitHub.psm1
+    if ( -not (Test-HDFreeSpace -Size 50)) {
```

Comment:
These tests don't need 50 GB, 1 GB should be plenty _by far_.

Reply:
```suggestion
    if ( -not (Test-HDFreeSpace -Size 5)) {
```